state/storage.go

```diff
@@ -178,6 +178,14 @@ func (s *Storage) GlobalSnapshot() (ss StartupSnapshot, err error) {
 
 // Extract hero info for all rooms. Requires a prepared snapshot in order to be called.
 func (s *Storage) MetadataForAllRooms(txn *sqlx.Tx, tempTableName string, result map[string]internal.RoomMetadata) error {
+	loadMetadata := func(roomID string) internal.RoomMetadata {
+		metadata, ok := result[roomID]
+		if !ok {
+			metadata = *internal.NewRoomMetadata(roomID)
+		}
+		return metadata
```

Comment:
Should we write this back to `result`? I'm leaning towards "no": I suppose it doesn't make much difference since we ought to have `_, ok := ...` checks elsewhere, and it might give the false impression that we know about this room.

Reply:
No, leave it to the caller.